packages/core/database/lib/fields.js

```diff
@@ -118,16 +120,23 @@ const parseTime = value => {
 };
 
 const parseDate = value => {
-  if (dateFns.isDate(value)) return dateFns.format(value, 'yyyy-MM-dd');
-  try {
-    let date = dateFns.parseISO(value);
+  const found = _.isString(value) ? value.match(partialDateRegex) : [];
+  const extractedValue = found[0];
 
-    if (dateFns.isValid(date)) return dateFns.format(date, 'yyyy-MM-dd');
+  if (extractedValue && !dateRegex.test(value)) {
+    // TODO V5: throw an error when format yyyy-MM-dd is not respected
+    // throw new InvalidDateError(`Invalid format, expected yyyy-MM-dd`);
+    process.emitWarning(
+      `[deprecated] Using a date format other than yyyy-MM-dd will be deprecated. Use format yyyy-MM-dd instead. Value found: ${value}. Date kept: ${extractedValue}.`
```

Comment:
```suggestion
      `[deprecated] Using a date format other than YYYY-MM-DD will be removed in future versions. Use format YYYY-MM-DD instead. Value found: ${value}. Date kept: ${extractedValue}.`
```

Reply:
I like Alex's suggestions and would like to suggest on top of it:

```suggestion
      `[deprecated] Using a date format other than YYYY-MM-DD will be removed in future versions. Date received: ${value}. Date stored: ${extractedValue}.`
```